apps/web/app/utils/extract-from-formdata.ts

```diff
@@ -13,3 +13,66 @@ export function extract<T extends ZodTypeAny>(
   }
   return parsed.data;
 }
+
+export type ParseFromDataRes<T extends ZodObject<any>> =
+  | {
+      success: true;
+      data: z.infer<T>;
+      error: null;
+    }
+  | {
+      success: false;
+      data: null;
+      error: ZodError;
+    };
+
+export function parseFormData<T extends ZodObject<any>>(
+  formData: FormData,
+  parser: T
+): ParseFromDataRes<T> {
+  const keys = Object.keys(parser.shape);
+
+  const data: Record<string, any> = {};
```

Comment:
Using `unknown` instead of `any` is better in many cases, as `any` completely disables type checking and can expand to unwanted places.

```suggestion
  const data: Record<string, unknown> = {};
```